src/Build/BackEnd/Components/ProjectCache/ProjectCacheService.cs

```diff
@@ -218,18 +234,53 @@ private IEnumerable<ProjectCacheDescriptor> GetProjectCacheDescriptors(ProjectIn
                     ProjectCacheException.ThrowForErrorLoggedInsideTheProjectCache("ProjectCacheInitializationFailed");
                 }
 
-                return new ProjectCachePlugin(pluginTypeName, pluginInstance);
+                FileAccessManager.HandlerRegistration handlerRegistration = _fileAccessManager.RegisterHandlers(
+                    (buildRequest, fileAccessData) =>
+                    {
+                        // TODO: Filter out projects which do not configure this plugin
```

Comment:
Required? Something that will actually get done?

Reply:
It's a bit of a niche scenario so not something I've prioritized. If two projects configured different plugins, both would get accesses from both projects, but only be queried for the projects which configured it. Any reasonable plugin would just ignore the extra stuff, and because this is such an edge case I'm not considering this a hard requirement to fix, but ideally I'd like to tighten this up at some point.